### PR TITLE
test: tests for component FaveStar

### PR DESCRIPTION
**Pull request by yardz**
### SUMMARY
- moving `FaveStar.tsx` to `FaveStar` folder
- switching FaveStar to function component
- tests for FaveStar component

### TEST PLAN
No change in behavior. All tests must pass.

### CODE REVIEW

File: superset-frontend/src/components/FaveStar/FaveStar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    showTooltip: false,
+  };
+
+  const { rerender } = render(<FaveStar {...props} />);
+  expect(props.fetchFaveStar).toBeCalledTimes(1);
+  expect(props.fetchFaveStar).toBeCalledWith(props.itemId);
+
+  rerender(<FaveStar {...{ ...props, itemId: 2 }} />);
+  expect(props.fetchFaveStar).toBeCalledTimes(1);
+  // expect(screen.getByText('backend-name')).toBeInTheDocument();
```
Comment (geido) [inline superset-frontend/src/components/FaveStar/FaveStar.test.tsx:104]: Seems like you forgot to remove a comment

**Review by geido — APPROVED** [review]
LGTM!

File: superset-frontend/src/components/FaveStar/FaveStar.test.tsx (hunk 1 above)
Comment (yardz) [reply superset-frontend/src/components/FaveStar/FaveStar.test.tsx]: Tnks!

**Review by pkdotson — APPROVED** [review]
LGTM! thanks for changing to typescript!

**Pull request merged**